api/envoy/extensions/filters/http/thrift_to_metadata/v3/thrift_to_metadata.proto

```diff
@@ -0,0 +1,133 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.thrift_to_metadata.v3;
+
+import "envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto";
+
+import "google/protobuf/struct.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.thrift_to_metadata.v3";
+option java_outer_classname = "ThriftToMetadataProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/thrift_to_metadata/v3;thrift_to_metadatav3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Thrift-To-Metadata Filter]
+//
+// The configuration for transforming thrift metadata into filter metadata. This is useful
+// for matching load balancer subsets, logging, etc.
+//
+// Thrift to Metadata :ref:`configuration overview <config_http_filters_thrift_to_metadata>`.
+// [#extension: envoy.filters.http.thrift_to_metadata]
+
+enum Fields {
+  // The Thrift method name, string value.
+  METHOD_NAME = 0;
+
+  // The Thrift protocol name, string value. Values are "binary", "binary/non-strict", and "compact".
+  PROTOCOL = 1;
+
+  // The Thrift transport name, string value. Values are "framed", "header", and "unframed".
+  TRANSPORT = 2;
+
+  // The Thrift message type, singed 16-bit integer value.
+  HEADER_FLAGS = 3;
+
+  // The Thrift sequence ID, singed 32-bit integer value.
+  SEQUENCE_ID = 4;
+
+  // The Thrift message type, string value. Values in request are "call" and "oneway", and in response are "reply" and "exception".
+  MESSAGE_TYPE = 5;
+
+  // The Thrift reply type, string value. This is only valid for response rules. Values are "success" and "error".
+  REPLY_TYPE = 6;
+}
+
+message KeyValuePair {
+  // The namespace — if this is empty, the filter's namespace will be used.
+  string metadata_namespace = 1;
+
+  // The key to use within the namespace.
+  string key = 2 [(validate.rules).string = {min_len: 1}];
```

Comment:
IIRC the metadata allows nesting of keys. Is that supported here? If yes, please document how (I assume period-delimited keys). It not, should it be?

Reply:
hmm i have little knowledge on nesting of keys here but I was following prior art, e.g., http/header_to_metadata filter.

https://github.com/envoyproxy/envoy/blob/9536ce2d3fdb234f8006ba4092e59577c15ea179/api/envoy/config/core/v3/base.proto#L248

Here's is the namespace to struct mapping, and we have `key` `value` mapping inside of the struct.

Reply:
Ok. I suppose in this case it's probably better to be consistent.